lib/ransack/adapters/active_record/base.rb

```diff
@@ -45,7 +45,7 @@ def ransackable_attributes(auth_object = nil)
         # For overriding with a whitelist array of strings.
         #
         def ransackable_associations(auth_object = nil)
-          reflect_on_all_associations.map { |a| a.name.to_s }
+          @ransackable_associations ||= reflect_on_all_associations.map { |a| a.name.to_s }
```

Comment:
Yeah, I hesitated to make this change. This PR isn't meant to be mergeable in its current state, I just went through the code and applied optimizations to the code that the report highlighted.

Now that we have this proof-of-concept, we can discuss which changes make sense.